### PR TITLE
feat: [CDS-77529]: implement webhooks events page 

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

This PR takes care of the following items :- 

- Setting up the events page UI layout with zero state entities
- Setting up the individual details row for each event
- Setting up the filters required on the events page
- Setting up the right drawer for the payload details

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/d138070d-1d39-4541-a06c-6f8cc515bc98

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by Eric761 — APPROVED** [review]

File: src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx [hunk 1]
```diff
@@ -5,8 +5,158 @@
...
+        breadcrumbs={<NGBreadcrumbs customPathParams={{ module }} />}
+        className={css.header}
+        content={<WebhooksTabs />}
+      />
+      <Page.SubHeader className={css.toolbar}>
+        <div>SubHeader</div>
+      </Page.SubHeader>
+      <div className={css.content}>
+        {state === STATUS.error && (
+          <Page.Error message={getErrorInfoFromErrorObject(defaultTo(error, {}) as any)} onClick={refetch as any} />
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx:114]: defaultTo for error should not be needed and `any` typecasting as well

**Review by PratyushGar — APPROVED** [review]

**Comment by vikrant-harness** [comment]
This is in development feature , Coverage will be added with this ticket :- https://harness.atlassian.net/browse/CDS-78807

File: src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx:114]: the error type for the openAPI API's are unknown types hence the typecasting , defaultTo just a safety net if the API fails without error response message , will remove in next iteration

File: src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx [hunk 2]
```diff
@@ -5,8 +5,158 @@
...
+import EmptyContentImg from '@common/images/EmptySearchResults.svg'
+import NoData from '@cd/components/EnvironmentsV2/PageTemplate/NoData'
+import WebhooksTabs from '../WebhooksTabs'
+import WebhooksEventsList from './WebhooksEventsList'
+import css from '../Webhooks.module.scss'
 
 export default function WebhookEvents(): JSX.Element {
-  return <div>Events Page</div>
+  const { getString } = useStrings()
+  const { module } = useParams<ProjectPathProps & ModulePathParams>()
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx:43]:
```suggestion
  const { module } = useParams<ModulePathParams>()
```

File: src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx [hunk 3]
```diff
@@ -5,8 +5,158 @@
...
+    error,
+    refetch
+  } = useListGitxWebhooksQuery({
+    queryParams: {
+      limit: size,
+      page: page ? page - 1 : 0
+    }
+  })
+
+  enum STATUS {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx:62]: Do we need to define this here? DO we not have this enum/interface/union type ?

File: src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx]: done

File: src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhookEvents/WebhookEvents.tsx]: extracted in a util and used both in listing and events page

**Review by pankajsingh-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Comment by vikrant-harness** [comment]
Coverage will be taken up as a separate effort including the lint warnings for one part , Jest is failing for unrelated tests CI team is working on the same

**Review by Eric761 — APPROVED** [review]

File: src/modules/75-cd/pages/webhooks/WebhookEvents/PayloadDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { Drawer } from '@blueprintjs/core'
```
[the remark this answers is not in the document]
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhookEvents/PayloadDetails.tsx:9]: sure

**Review by harshilbits — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**